src/models/StyledNativeComponent.js

```diff
@@ -42,8 +43,12 @@ export default (constructWithOptions: Function, InlineStyle: Function) => {
 
       this.attrs = Object.keys(attrs).reduce((acc, key) => {
         const attr = attrs[key]
+        const extendsReactComponent = hasInInheritanceChain(attr, Component)
```

Comment:
let's inline this, no need to do the work unless `attr` is confirmed to be a function

Reply:
updated